dmscripts/email_engine/typing.py

```diff
@@ -61,6 +63,23 @@ def from_str(cls, s: str) -> "EmailNotification":
         """Parse a dict literal representation of a notification"""
         return cls(**literal_eval(s))
 
+    @property
+    def sha256_hash(self) -> str:
```

Comment:
Why do we need to change the way we create the reference? I thought we already had `dmutils.email.DMNotifyEmail.get_reference`?

Reply:
The email engine doesn't use `dmutils.email.DMNotifyEmail`, it goes straight to the Notify client. See https://github.com/alphagov/digitalmarketplace-scripts/blob/main/dmscripts/email_engine/__init__.py#L130

Reply:
I wonder if we could use the one from dmutils though, let me check for the differences

Reply:
Ok, I think I have it working with `dmutils.email.DMNotifyEmail` and the hash method we already use. Just need to sort out some tests